Sources_v3/StreamChat/Query/Sorting/ChannelListSortingKey.swift

```diff
@@ -43,3 +47,15 @@ extension ChannelListSortingKey {
         .init(key: rawValue, ascending: isAscending)
     }
 }
+
+/// In case elements have same or `nil` target sorting values it's not possible to guarantee that order of elements will be the same
+/// all the time. So we need to additionally provide safe sorting option.
+extension Array where Element == Sorting<ChannelListSortingKey> {
+    func appendingSafeSortingIfNeeded() -> [Sorting<ChannelListSortingKey>] {
+        guard !contains(where: { $0.key == .cid }), !isEmpty else {
+            return self
+        }
```

Comment:
Do we really need `isEmpty == false` check? Shouldn't the `cid` sorting key be added if the array is empty?

Reply:
AFAIK backend will sort them "safely" if client sends an empty sorting

Reply:
Yep, @b-onc is right.
Backend will apply default sorting with secondary by `cid` in this case.